tfjs-core/src/ops/conv_util.ts

```diff
@@ -365,24 +365,23 @@ function computeOutputShape2D(
 }
 
 function computeOutputShape4D(
-    inShape: [number, number, number, number], fieldSize: number,
-    outChannels: number, stride: number, zeroPad?: number,
+    inShape: [number, number, number, number],
+    filterShape: [number, number, number], outChannels: number,
+    strides: [number, number, number], zeroPad?: number,
     roundingMode?: 'floor'|'round'|'ceil'): [number, number, number, number] {
   if (zeroPad == null) {
-    zeroPad = computeDefaultPad(inShape, fieldSize, stride);
+    zeroPad = computeDefaultPad(inShape, filterShape[0], strides[0]);
   }
-  const inputDepth = inShape[0];
-  const inputRows = inShape[1];
-  const inputCols = inShape[2];
-
-  const outputDepths =
-      round((inputDepth - fieldSize + 2 * zeroPad) / stride + 1, roundingMode);
-  const outputRows =
-      round((inputRows - fieldSize + 2 * zeroPad) / stride + 1, roundingMode);
-  const outputCols =
-      round((inputCols - fieldSize + 2 * zeroPad) / stride + 1, roundingMode);
-
-  return [outputDepths, outputRows, outputCols, outChannels];
+  const outShape: [number, number, number, number] = [0, 0, 0, outChannels];
+  for (let index = 0; index < 3; index++) {
+    if (inShape[index] + 2 * zeroPad >= filterShape[index]) {
+      outShape[index] = round(
+          (inShape[index] - filterShape[index] + 2 * zeroPad) / strides[index] +
+              1,
+          roundingMode);
+    }
+  }
+  return outShape;
```

Comment:
Adds two changes in `computeOutputShape4D` function:

1. add a if-branch to make sure all dimensions of outShape are non-negative. 
2. outputDepths, outputRows, outputCols should be computed from the corresponding strides and filterSize respectfully.